pytorch_lightning/loops/epoch/training_epoch_loop.py

```diff
@@ -47,8 +52,8 @@ def __init__(self, min_steps: int, max_steps: int):
         self.batch_progress = Progress()
         self.scheduler_progress = SchedulerProgress()
 
-        self.batch_loop = TrainingBatchLoop()
-        self.val_loop = loops.EvaluationLoop()
+        self.batch_loop = Optional[TrainingBatchLoop]
+        self.val_loop = Optional["loops.EvaluationLoop"]
```

Comment:
Optional on the right hand side??

Did you mean?

```suggestion
        self.batch_loop: Optional[TrainingBatchLoop] = None
        self.val_loop: Optional["loops.EvaluationLoop"] = None
```

Regardless, why make it Optional? Not optional has the advantage of not having to deal with the `None` checks

Reply:
> Regardless, why make it Optional? Not optional has the advantage of not having to deal with the None checks

yes but mypy apparently needs the annotation if we set it to None.
and we can't set a default loop here, as we can't deal with initialization beforehand.